### PR TITLE
Drop puppet bats

### PR DESCRIPTION
**Pull request by ezr-ondrej**
The proxy assignment to taxonomies is not puppet related, so I'm moving
it to the Foreman bats.

We are planning to drop puppet from Foreman, so Foreman and Katello
pipelines should not test puppet anymore.

We could use this to add Puppet plugin pipeline or general plugin
pipeline in the future, so I'm leaving the bats itself.

### CODE REVIEW

File: bats/fb-test-foreman.bats [hunk 1]
```diff
@@ -16,3 +18,11 @@ set -o pipefail
 @test "check smart proxy is registered" {
   hammer proxy info --name=$(hostname -f)
 }
+
+@test "Assign proxy to default taxonomies" {
```
Comment (ekohl) [inline bats/fb-test-foreman.bats:22]: This looks like it doesn't matter anymore and we should actually drop this workaround on 1.23+

**Review by ekohl — CHANGES_REQUESTED** [review]
This isn't really complete because that's not the nightly pipeline. That playbook is a leftover from old stuff and we should probably delete it.

Here is the Foreman pipeline: https://github.com/theforeman/forklift/blob/daa523e6d5c1a3ad2696fede8fc02b6052c03234/pipelines/vars/forklift_foreman.yml#L14

For our plugin pipeline it's here: https://github.com/theforeman/forklift/blob/daa523e6d5c1a3ad2696fede8fc02b6052c03234/pipelines/vars/forklift_plugins.yml#L32

Katello doesn't contain it: https://github.com/theforeman/forklift/blob/master/pipelines/vars/forklift_katello.yml

That means it uses the default (which you did modify). Removing it there means we also don't test it on stable releases anymore.

Luna is the same as Katello and located here: https://github.com/theforeman/forklift/blob/master/pipelines/vars/forklift_luna.yml

Here is an example of how to make something version dependent:
https://github.com/theforeman/forklift/blob/daa523e6d5c1a3ad2696fede8fc02b6052c03234/pipelines/vars/forklift_katello.yml#L11

Note that Foreman and Plugins use Foreman version numbering while Katello and Luna use Katello version numbering (2.5 vs 4.0).

**Comment by ezr-ondrej** [comment]
Thanks @ekohl for the guidance :) updated the vars and added the bats to plugins pipeline.
Once nightlies are fixed, I'll scrach build and test the pipelines :)

File: pipelines/vars/forklift_plugins.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ server_box:
         - "--enable-foreman-plugin-bootdisk"
         - "--enable-foreman-plugin-discovery"
         - "--enable-foreman-plugin-hooks"
+        - "--enable-foreman-plugin-puppet"
```
Comment (ekohl) [inline pipelines/vars/forklift_plugins.yml:20]: This should also be conditional so we don't break pipes like https://ci.theforeman.org/job/foreman-plugins-2.5-rpm-test-pipeline/ and https://ci.theforeman.org/job/foreman-plugins-2.5-deb-test-pipeline/ (and older).

File: bats/fb-test-foreman.bats [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
 set -o pipefail
 
+load foreman_helper
+
```
Comment (ekohl) [inline bats/fb-test-foreman.bats:7]: Looks like a leftover

**Review by ekohl — CHANGES_REQUESTED** [review]

**Review by ekohl — COMMENTED** [review]
I still think the katello and luna pipelines need to enable it on run the Puppet tests on Katello <= 4.1

File: pipelines/vars/forklift_plugins.yml [hunk 4]
```diff
@@ -17,6 +17,7 @@ server_box:
         - "--enable-foreman-plugin-bootdisk"
         - "--enable-foreman-plugin-discovery"
         - "--enable-foreman-plugin-hooks"
+        - "{{ '--enable-foreman-plugin-puppet' if pipeline_version == 'nightly' or pipeline_version is version('3.0', '>=') else '' }}"
```
Comment (evgeni) [inline pipelines/vars/forklift_plugins.yml:20]:
```suggestion
        - "{{ '--enable-foreman-plugin-puppet' if pipeline_version == 'nightly' or pipeline_version is version('3.1', '>=') else '' }}"
```

Or does it really matter? The plugin is already installed on 3.0, and doesn't need manual installation.

File: playbooks/bats_pipeline_foreman_nightly.yml [hunk 5]
```diff
@@ -11,7 +11,6 @@
     bats_tests:
       - "fb-verify-packages.bats"
       - "fb-test-foreman.bats"
-      - "fb-test-puppet.bats"
```
Comment (evgeni) [inline playbooks/bats_pipeline_foreman_nightly.yml:14]: this file is gone.

**Comment by evgeni** [comment]
> I still think the katello and luna pipelines need to enable it on run the Puppet tests on Katello <= 4.1

this can be done here:
https://github.com/theforeman/forklift/blob/9563ec314184e7cb8f392aa1c6c94e90a96d3664/pipelines/vars/forklift_katello.yml#L16-L18
and here:
https://github.com/theforeman/forklift/blob/9563ec314184e7cb8f392aa1c6c94e90a96d3664/pipelines/vars/forklift_luna.yml#L46-L50

yes, making the array entry an empty string works these days (since https://github.com/theforeman/forklift/pull/1415)

File: pipelines/vars/forklift_plugins.yml (hunk 4 above)
Comment (ezr-ondrej) [reply pipelines/vars/forklift_plugins.yml:20]: It was for not installing it prior 3.0 release... now it does not matter :)

File: pipelines/vars/forklift_foreman.yml [hunk 6]
```diff
@@ -11,7 +11,6 @@ server_box:
       bats_tests:
         - "fb-verify-packages.bats"
         - "fb-test-foreman.bats"
-        - "fb-test-puppet.bats"
```
Comment (ezr-ondrej) [inline pipelines/vars/forklift_foreman.yml:14]: I guess we can just ignore that this is kinda relevant to current releases?

File: pipelines/vars/forklift_luna.yml [hunk 7]
```diff
@@ -30,6 +30,9 @@ server_box:
         - "--enable-foreman-plugin-openscap"
         - "--enable-foreman-proxy-plugin-openscap"
 
+        - "--enable-foreman-cli-puppet"
+        - "--enable-foreman-plugin-puppet"
```
Comment (ezr-ondrej) [inline pipelines/vars/forklift_luna.yml:34]: Do I need to care that these are not available on 2.5 release? Are we running this pipeline there?

File: pipelines/vars/forklift_plugins.yml (hunk 4 above)
Comment (ezr-ondrej) [reply pipelines/vars/forklift_plugins.yml]: or maybe not, lets be safe :)

File: pipelines/vars/forklift_luna.yml (hunk 7 above)
Comment (ezr-ondrej) [reply pipelines/vars/forklift_luna.yml]: I did to be on the safe side

Comment (evgeni) [reply pipelines/vars/forklift_luna.yml]: we only *run* luna for nightly, but there is nothing that forbids a user to run this with `pipeline_version` parameter, so thanks!

File: pipelines/vars/forklift_foreman.yml (hunk 6 above)
Comment (evgeni) [reply pipelines/vars/forklift_foreman.yml:14]: No?

Just do:
```yaml
- "{{ 'fb-test-puppet.bats' if pipeline_version != 'nightly' and pipeline_version is version('3.1', '<') else '' }}"

File: bats/fb-test-puppet.bats [hunk 8]
```diff
@@ -70,14 +70,6 @@ fi
   hammer puppet-environment update --name=production --locations "Default Location" --organizations "Default Organization"
 }
 
-@test "Assign proxy to default taxonomies" {
```
Comment (ekohl) [inline bats/fb-test-puppet.bats:73]: Let's separate this out to its own PR. I think that can already be merged and reduces the scope of this PR.

**Review by ekohl — COMMENTED** [review]
For completeness: at least now the installer enables the plugin by default. While this work is needed for https://github.com/theforeman/foreman-installer/pull/717, we're some way away from completing that.

**Comment by ekohl** [comment]
> This will make Katello not run Puppet tests either, which I think is wrong for pre 4.2?

Today it must continue testing it in all branches, including nightly. It's still enabled by default.

File: bats/fb-test-puppet.bats (hunk 8 above)
Comment (ezr-ondrej) [reply bats/fb-test-puppet.bats]: extracted to #1419
